pkg/daemon/daemon.go

```diff
@@ -431,13 +435,19 @@ func (dn *Daemon) applyNodePtpProfile(runID int, nodeProfile *ptpv1.PtpProfile)
 				glog.Errorf("Error creating named pipe, GNSS monitoring will not work as expected %s", e.Error())
 			}
 			gpsDaemon := &gpsd{
-				name:        GPSD_PROCESSNAME,
-				execMutex:   sync.Mutex{},
-				cmd:         nil,
-				serialPort:  GPSD_SERIALPORT,
+				name:      GPSD_PROCESSNAME,
+				execMutex: sync.Mutex{},
+				cmd:       nil,
+				serialPort: func() string {
+					if output.gnssSerialPort == "" {
+						return GPSDDefaultGNSSSerialPort
```

Comment:
I would just set output.gnssSerialPort to the default when initializing the ptp4lConf right before the call to populatePtp4lConf, and then it would be overridden if set.  No need to check here then.

Reply:
done